File: ticket-backend/src/main/java/aics/domain/movie/models/MovieModel.java
package aics.domain.movie.models;

import aics.domain.movie.Movie;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.Base64;

@Data
@Accessors(chain = true)
public class MovieModel implements Serializable {
    private Long movieId;
    private String name;
    private String description;
    private String image;
    private String directors;
    private String script;
    private String actors;
    private String appropriateness;
    private int duration;

    public static MovieModel fromMovie(Movie movie) {
        if (movie == null) {
            return null;
        }
        return new MovieModel()
            .setMovieId(movie.getMovieId())
            .setName(movie.getName())
            .setDescription(movie.getDescription())
            .setImage(Base64.getEncoder().encodeToString(movie.getImage()))
            .setDirectors(movie.getDirectors())
            .setScript(movie.getScript())
            .setActors(movie.getActors())
            .setAppropriateness(movie.getAppropriateness())
            .setDuration(movie.getDuration());
    }
}

File: ticket-frontend/src/root/shared/models/movie-model.ts
import { TypeUtils } from '../../../modules/core/type-utils';

export class MovieModel {
    movieId: number = 0;
    name: string = '';
    description: string | null = null;
    image: string = '';
    directors: string = '';
    script: string = '';
    actors: string = '';
    appropriateness: string = '';
    duration: number = 0;

    static fromObj(obj: any): MovieModel | null {
        if (!obj) {
            return null;
        }
        const movieModel: MovieModel = new MovieModel();
        movieModel.movieId = obj.movieId;
        movieModel.name = obj.name;
        movieModel.description = obj.description;
        movieModel.image = obj.image;
        movieModel.directors = obj.directors;
        movieModel.script = obj.script;
        movieModel.actors = obj.actors;
        movieModel.appropriateness = obj.appropriateness;
        movieModel.duration = obj.duration;
        return movieModel;
    }

    static listFromObjList(objs: any[]): MovieModel[] {
        if (!objs) {
            return [];
        }
        return objs.map(MovieModel.fromObj).filter(TypeUtils.isNonNullable);
    }
}